### DISCUSSION TITLE
how to import external library into boot file and use external library.

### DISCUSSION THREAD

**Discussion by micro155** [quasarframework/quasar#14903] [General - Components / Directives / etc]
Hi!  I'm making a google sign-in feature using Vue 3 and Quasar Framework in my app.

Using the 'vue3-google-oauth2' npm package, I’m trying to call $gAuth.signIn() after initializing through Quasar boot files.

However, when called $gAuth.signIn(), false is returned. So google oauth is not running.

How can I use 'vue3-google-oauth2' library in Quasar with Quasar boot files?

this is my code

googleoauth.js
```
import { boot } from 'quasar/wrappers';
import GAuth from 'vue3-google-oauth2';

let app;
export default boot(({ app }) => {
  const gAuthOption = {
    clientId: '835170725866-nt6qibip4tb7jafav2k20j83ipso67mo.apps.googleusercontent.com',
    scope: 'profile email',
    prompt: 'select_account',
  };
  app.use(GAuth, gAuthOption);

  app = app;
});

export { app };
```

quasar.config.js
```
...
boot : ['googleoauth']
...
```

LoginPage.vue <------ I want to make google signin feature in this file
```
...
import { getCurrentInstance } from 'vue';

const app = getCurrentInstance();


app.appContext.config.globalProperties.$gAuth.signIn(); // return false (not working)
...
```

- References
vue3-google-oauth2 : https://www.npmjs.com/package/vue3-google-signin

**Comment by MilosPaunovic** [comment]
Why don't you, for starters, try to name your variable differently and see if that resolves the issue:

```js
let app;
export default boot(({ app }) => {
```
